README.md

```diff
@@ -89,58 +92,275 @@ Now you can import like this:
 const zbench = @import("zbench");
 ```
 
+### Compatibility Notes
+
+***Supported Version***: As of now, this fork of zBench is tested and supported on Zig version ***0.12.0-dev.1849+bb0f7d55e***.
+
 ## Usage
-Create a new benchmark function in your Zig code. This function should take a single argument of type *zbench.Benchmark. The function would run the code you wish to benchmark.
+The main type in this library is the `Benchmark` which contains the state for a single benchmark at any given time. A single instance can (and should) run several
+benchmarks, just not concurrently. You can make use of the timing functionality in `Benchmark` to get timings directly, or you can pass a bench-runner to the `run`
+function. The latter is the simplest, and fits most usecases. Here are some examples:
 
+### Standalone function
+The simplest case is when you wish to benchmark a standalone function
 ```zig
-fn benchmarkMyFunction(b: *zbench.Benchmark) void {
+fn myBenchRunner() void {
     // Code to benchmark here
 }
 ```
-You can then run your benchmarks in a test:
+You can then run your benchmarks in either a test, or main as an executable. The latter is preferable as tests can generate noise and obscure the benchmark output.
+Next we instantiate a `Benchmark` instance.
 ```zig
-test "bench test" {
-    var allocator = std.heap.page_allocator;
-    var b = try zBench.Benchmark.init("benchmarkMyFunction", &allocator);
-    try zBench.run(benchmarkMyFunction, &b);
+pub fn main() !void {
+    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
+ 
+    const second: u64 = 1_000_000_000;
+    const bench_iterations: u64 = 128;
+
+    var bench = try zbench.Benchmark.init(second, bench_iterations, gpa.allocator());
+    defer bench.deinit();
 }
 ```
+The first argument to `init` is an estimate in nanoseconds for the maximum amount of time we are willing to wait for any given benchmark to finish,
+the second argument is the maximum number of repetitions or runs for each benchmark. Now to perform the benchmark we pass `myBenchRunner` to `bench.run`,
+and the name of our benchmark. The complete example looks like this:
+```zig
+const std = @import("std");
+const zbench = @import("zbench");
 
-### Compatibility Notes
-Zig is in active development and the APIs can change frequently, making it challenging to support every dev build. This project currently aims to be compatible with stable, non-development builds to provide a consistent experience for the users.
+fn myBenchRunner() void {
+    // Code to benchmark here
+}
 
-***Supported Version***: As of now, zBench is tested and supported on Zig version ***0.11.0***.
+pub fn main() !void {
+    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
 
-### Benchmark Functions
-Benchmark functions have the following signature:
+    const second: u64 = 1_000_000_000;
```

Comment:
is this the max. duration in ns ? should name it `max_duration_ns` or similar if so.